plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -432,6 +432,10 @@ public List<String> getAllTables(String databaseName)
             return ImmutableList.of();
         }
         catch (AmazonServiceException e) {
+            if (!getAllDatabases().contains(databaseName)) {
```

Comment:
I think I would do this differently. Would something more like this work? Add on to the catch above

```
catch (EntityNotFoundException | AccessDeniedException e) {
    return ImmutableList.of();
}
```